### ISSUE TITLE
io.cache not working if writing after the end of the file size

### ISSUE THREAD

**Issue by radare** [radare/radare2#2739]
```
echo ji > x
r2 -w -e io.cache=true x
> w hello
> x
- offset -   0 1  2 3  4 5  6 7  8 9  A B  C D  E F  0123456789ABCDEF
0x00000000  6865 6cff ffff ffff ffff ffff ffff ffff  hel.............
```

**Comment by radare** [comment]
can we have a test for this @Maijin ?

**Comment by Maijin** [comment]
https://github.com/radare/radare2-regressions/commit/511a06636c1b05b9cc7e2046d9f292d69c1d09a0

**Comment by radare** [comment]
@condret is this fixed in siol?

**Comment by condret** [comment]
@radare i think it is

**Comment by radare** [comment]
@condret can you confirm?

**Comment by MaskRay** [comment]
Writing after the end should be forbidden but sadly it is accepted by io.cache=1 now, though `wci` does not work.

**Comment by radare** [comment]
i did omm to allocate a new map from an fd with the updated size.. but maps fuckup the whole thing when resizing happens. the ‘r’ command resizes the file, but doesnt updates the map if any.. which is pretty bad

**Comment by radare** [comment]
i filled an issue for that

**Comment by MaskRay** [comment]
Can be closed?

commit c0e036dcb0c6820f1cf03b4deb60804b3858ee35
Fix #8498 r command should resize the map (#8518)

**Comment by Maijin** [comment]
Thanks !
